registry/handlers/manifests.go

```diff
@@ -439,6 +458,13 @@ func (imh *manifestHandler) applyResourcePolicy(manifest distribution.Manifest)
 		default:
 			return errcode.ErrorCodeDenied.WithMessage("unknown manifest class for " + m.Config.MediaType)
 		}
+	case *ociartifact.DeserializedManifest:
+		switch m.MediaType {
+		case v1.MediaTypeArtifactManifest:
+			class = artifactClass
+		default:
+			return errcode.ErrorCodeDenied.WithMessage("unknown manifest class for " + m.MediaType)
+		}
```

Comment:
The `m.MediaType` is always `v1.MediaTypeArtifactManifest`.
```suggestion
		class = artifactClass
```

Reply:
changed accordingly.